src/simplifyAST.js

```diff
@@ -2,7 +2,7 @@ function deepMerge(a, b) {
   Object.keys(b).forEach(function (key) {
     if (['fields', 'args'].indexOf(key) !== -1) return;
 
-    if (a[key] && b[key]) {
+    if (a[key] && b[key] && typeof a[key] === 'object') {
```

Comment:
I figure if a[key] is not an object, merge should replace it with b[key],
regardless of what b[key] is.

On April 20, 2016 at 12:32:49 AM, Mick Hansen (notifications@github.com)
wrote:

> In src/simplifyAST.js
> <https://github.com/mickhansen/graphql-sequelize/pull/218#discussion_r60361299>
> :
>
> > @@ -2,7 +2,7 @@ function deepMerge(a, b) {
> >    Object.keys(b).forEach(function (key) {
> >      if (['fields', 'args'].indexOf(key) !== -1) return;
> >
> > -    if (a[key] && b[key]) {
> > +    if (a[key] && b[key] && typeof a[key] === 'object') {
>
> Probably should be prudent and check b[key] aswell just in case.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/mickhansen/graphql-sequelize/pull/218/files/96627513e74489da4d34388d73f61fa516636372#r60361299>
>